### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by scalalang2**
# Description

I put a Dockerfile to compile polygon-edge with docker.
This can improve the development productivity as we can create a sample [polygon-edge network locally](https://github.com/scalalang2/polygon-edge-docker-compose).

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

No Breaking changes

# Checklist

- [x] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

let's execute below command

```shell
docker build -t polygon-edge .
```

### CODE REVIEW

**Comment by scalalang2** [comment]
I have read the CLA Document and I hereby sign the CLA

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+COPY go.mod .
+COPY go.sum .
+
+RUN apk add --no-cache gcc librdkafka-dev libc-dev
+RUN cd /src && go mod download
+
+FROM build_base AS builder
+
+ADD . /src
+RUN cd /src && go build --tags musl -o polygon-edge main.go
```
Comment (epikichi) [inline Dockerfile:14]: Can we include the necessary `-ldflags`  to inject versioning context into the image builds as well? We can include them as `--build-args` if needed. 
- `-X "github.com/0xPolygon/polygon-edge/versioning.Version=${GITHUB_REF_NAME}"` 
- `-X "github.com/0xPolygon/polygon-edge/versioning.Commit=${GITHUB_SHA}"`

[Reference for flags](https://github.com/0xPolygon/polygon-edge/blob/f80b615edb8a69edea2a363b85a7f42c4a191251/.github/workflows/build.yml#L20)

**Review by epikichi — CHANGES_REQUESTED** [review]
Thanks for the contribution! This Dockerfile should help offer a familiar approach for those testing/developing in containerized environments. Added one change request to help provide additional context for developers while working with the image. 

@ZeljkoBenovic We have 3 separate Dockerfiles, including this one after acceptance. In the future, we can do something to [consolidate the one made for compose](https://github.com/0xPolygon/polygon-edge/blob/develop/docker/local/Dockerfile) with the proposed one in this PR, since they achieve similar goals.

File: Dockerfile (hunk 1 above)
Comment (scalalang2) [reply Dockerfile:14]: thanks for feedback : ) I'll apply those changes soon.

**Comment by ZeljkoBenovic** [comment]
Don't we already have [Dockerfile](https://github.com/0xPolygon/polygon-edge/blob/develop/docker/local/Dockerfile) in docker folder? In the same folder there is `docker-compose` as well.
Why is this `Dockerfile` different than the one we already have?

**Comment by scalalang2** [comment]
@ZeljkoBenovic 
At the time of writing this pull request, I couldn't find the `Dockerfile` since It's supposed to be on `root` directory.
There is no `Dockerfile` on root dir, so then, I thought that it doesn't exsit.

**Comment by scalalang2** [comment]
I agree that so I'll close this pr.

**Pull request closed without merging**